### DISCUSSION TITLE
Network requirements of Free5gc

### DISCUSSION THREAD

**Discussion by jgomezve** [Orange-OpenSource/towards5gs-helm#72] [Enhancements]
Hello

Why is it required to have static IP addresses for the SMF and AMF?
If is is about avoid IP change due to CNF crashes, wouldn't K8s services be a better option?
If it is about CNF discovery, wouldn't DNS and K8s Service discovery help here?

Also, why is it mandatory to have multiple interfaces? Cannot, for example, UPF and the CP-CNFs talk to the SMF using the same IP address?

I am highlighting these points, because Open5Gs, another 5GC doesn't seem to have these requirements

**Comment by raoufkh** [comment]
Hello

There are some limitations in the source code that impose fixed IP addresses, where additional interfaces are used. I will detail this and put it in this discussion to see which additional interfaces can be eliminated.

Bests
Abderaouf

**Comment by jgomezve** [comment]
Hi,
Just wondering.. If it is possible to run Free5Gc in Docker Compose, then it should be fine to run it on K8s without any additional IPs in the aforementioned NFs. 
Is my assumption right ?
